x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -209,12 +209,23 @@ public ActionRequestValidationException validate() {
             if (threadsPerAllocation < 1) {
                 validationException.addValidationError("[" + THREADS_PER_ALLOCATION + "] must be a positive integer");
             }
+            if (isPowerOf2(threadsPerAllocation) == false) {
+                validationException.addValidationError("[" + THREADS_PER_ALLOCATION + "] must be a power of 2");
+            }
             if (queueCapacity < 1) {
                 validationException.addValidationError("[" + QUEUE_CAPACITY + "] must be a positive integer");
             }
             return validationException.validationErrors().isEmpty() ? null : validationException;
         }
 
+        private static boolean isPowerOf2(int value) {
+            if (value == 0) {
+                return false;
+            }
+            double log2OfValue = Math.log(value) / Math.log(2);
+            return Math.floor(log2OfValue) == Math.ceil(log2OfValue);
```

Comment:
```suggestion
            return Integer.bitCount(value) == 1;
```

Avoids the need for conversion to floating point and expensive logarithm calculations.

Reply:
Clever!